### PR TITLE
Samw/element geometry builder params

### PR DESCRIPTION
**Pull request by swwilson-bsi**
Allow an app (including a two-process app) to specify an element's geometry in terms of an ElementGeometryDataEntry array, rather than a JSON geometry stream (or a fully cooked geomBinary). This allows the app to define the geometry that goes into the array using ElementGeometry.Builder (rather than GeometryStreamBuilder).

This PR should not be completed until the [PR of the same name](https://bentleycs.visualstudio.com/iModelTechnologies/_git/imodel02/pullrequest/218619) in ADO imodel02 is completed.

### CODE REVIEW

**Comment by bbastings** [comment]
Seems like we can get rid of IModelDb.elementGeometryUpdate? It's currently alpha...and it's more confusing. Also need to include ElementGeometryBuilderParams for GeometryPartProps.

**Comment by swwilson-bsi** [comment]
Problem ... Extract-API won't let me mark ElementGeometryBuilderParams as alpha and also make it a property of a public class such as GeometricElementProps. Everything about ElementGeometry, including ElementGeometryBuilderParams, is @alpha, and this new property is alpha. How can I get around this?

File: core/backend/src/Element.ts [hunk 1]
```diff
@@ -418,6 +418,8 @@ export abstract class GeometricElement extends Element implements GeometricEleme
   public category: Id64String;
   /** The GeometryStream for this GeometricElement. */
   public geom?: GeometryStreamProps;
+  /** How to build a GeometryStream. This is not a persistent property. It may be specified as an alternative to `geom` when inserting or updating an element.  */
+  public elementGeometryBuilderParams?: ElementGeometryUpdate;
```
Comment (pmconne) [inline core/backend/src/Element.ts:422]: Mark alpha

File: core/common/src/ElementProps.ts [hunk 2]
```diff
@@ -210,6 +213,8 @@ export interface GeometricElement2dProps extends GeometricElementProps {
  */
 export interface GeometryPartProps extends ElementProps {
   geom?: GeometryStreamProps;
+  /** How to build a GeometryStream. This is not a persistent property. It may be specified as an alternative to `geom` when inserting or updating an element.  */
```
Comment (pmconne) [inline core/common/src/ElementProps.ts:216]: Mark alpha.

File: core/backend/src/test/standalone/BinaryGeometryStreamWrite.test.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  const newCategoryCode = IModelTestUtils.getUniqueSpatialCategoryCode(dictionary, "ThisTestSpatialCategory");
+  const category = SpatialCategory.create(rwIModel, IModel.dictionaryId, newCategoryCode.value);
+  const spatialCategoryId = rwIModel.elements.insertElement(category);
+  category.setDefaultAppearance(new SubCategoryAppearance({ color: 0xff0000 }));
+  // const spatialCategoryId: Id64String = SpatialCategory.insert(rwIModel, IModel.dictionaryId, newCategoryCode.value!, new SubCategoryAppearance({ color: 0xff0000 }));
+
+  return { modelId, spatialCategoryId };
+}
+
+describe("BinaryGeometryStreamWriteTest", () => {
```
Comment (pmconne) [inline core/backend/src/test/standalone/BinaryGeometryStreamWrite.test.ts:40]: @swwilson-bsi do your new tests pass? Don't they require changes on the native side?

Comment (swwilson-bsi) [reply core/backend/src/test/standalone/BinaryGeometryStreamWrite.test.ts:40]: Yes, they require native code changes. I am running these tests locally (installnativeplatform.bat). 
This PR is supposed to be a branch off of the 'imodel02' branch.

Comment (pmconne) [reply core/backend/src/test/standalone/BinaryGeometryStreamWrite.test.ts:40]: Thanks. I have switched the target branch for your PR to imodel02.

**Review by pmconne — DISMISSED** [review]
Please mark `GeometricElement.elementGeometryBuildParams` as `@alpha` and rerun extract-api.

File: core/backend/src/Element.ts (hunk 1 above)
Comment (swwilson-bsi) [reply core/backend/src/Element.ts:422]: Done

Comment (pmconne) [reply core/backend/src/Element.ts:422]: Did you push that change? It is still missing a release tag.

Comment (swwilson-bsi) [reply core/backend/src/Element.ts:424]: I missed that case. Thanks for pointing it out.

**Review by pmconne — COMMENTED** [review]
Looks good to me but I defer to @bbastings.

File: core/backend/src/Element.ts [hunk 4]
```diff
@@ -418,6 +418,8 @@ export abstract class GeometricElement extends Element implements GeometricEleme
   public category: Id64String;
   /** The GeometryStream for this GeometricElement. */
   public geom?: GeometryStreamProps;
+  /** How to build the element's GeometryStream. This is used for insert and update only. It is not a persistent property. It will be undefined in the properties returned by functions that read a persistent element. It may be specified as an alternative to `geom` when inserting or updating an element.  */
+  public elementGeometryBuilderParams?: ElementGeometryUpdate;
```
[the remark this answers is not in the document]
Comment (swwilson-bsi) [reply core/backend/src/Element.ts]: oops! Fixed.

**Review by bbastings — APPROVED** [review]

File: common/api/core-backend.api.md [hunk 5]
```diff
@@ -1963,6 +1964,7 @@ export abstract class GeometricElement extends Element implements GeometricEleme
     static get className(): string;
     // @internal (undocumented)
     protected collectPredecessorIds(predecessorIds: Id64Set): void;
+    elementGeometryBuilderParams?: ElementGeometryUpdate;
```
Comment (bbastings) [inline common/api/core-backend.api.md:1967]: Re-run extract-api?

Comment (swwilson-bsi) [reply common/api/core-backend.api.md]: done

**Pull request merged**